src/components/SignIn.js

```diff
@@ -95,14 +182,23 @@ export default function SignIn() {
               </Link>
             </Grid>
             <Grid item>
-              <Link href="#" variant="body2">
+              <Link href="/register" variant="body2">
                 {"Don't have an account? Sign Up"}
               </Link>
             </Grid>
           </Grid>
         </Box>
       </Box>
-      <Copyright sx={{ mt: 8, mb: 4 }} />
+      {/* useLogin hook errors */}
+      {errors && (
+        <Snackbar
+          open={open}
+          autoHideDuration={6000}
+          onClose={handleClose}
+          message={errors.msg} // Use the error message from the errors object
+          action={action}
+        />
+      )}
```

Comment:
Is there a reason we have two Snackbar components in this container?

Reply:
Maybe I got it wrong at time, I thought because we have 2 errors one from useLogin hook another from handleLoginError function so the message will be different. I created 2 snackbar for that

Reply:
The one we process in the `handleLoginError` function is the same as what's returned from the `useLogin` hook. We just use that `handleLoginError` function so we can do more steps like, opening the snackbar and logging the error.  You can try it by doing a `console.log("Error from handleLoginError: ", error)` inside `handleLoginError` and then a `console.log("Error from hook: ", error)` under the `useLogin` Hook. 

In fact, in this case we can probably just leave line 96 as: `const { submit: signIn, loading, } = useLogin(formData, loginHookOptions);` since we don't need to use the raw errors directly in our code here. 

Also material-ui doesn't really allow two snackbars to be shown at the same time, one will overtake the other (though in this case it wouldn't have been noticeable since both snackbars would've had the same content anyway) : https://mui.com/material-ui/react-snackbar/#consecutive-snackbars

Reply:
An alternative option; would be to keep just the second snackbar, remove the `errors &&`, and make it only open when the errors is not null:

```
<Snackbar
  open={!!errors}
  autoHideDuration={6000}
  onClose={handleClose}
  message={errors?.msg}
  action={action}
/>
```

In this option, we would need to keep what you have currently where we get the errors from the hook's internal state: `const { submit: signIn, loading, errors } = useLogin(formData, loginHookOptions);`

And we would only use `handleLoginError` for logging but _not_ for setting any kind of UI state / showing/hiding the snackbar etc.

Reply:
Thank you for clarification, I update it